### PR TITLE
Add share networking code

### PR DESCRIPTION
**Pull request by manuelmarcos**

### CODE REVIEW

File: AlamofireRequest.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        let responseSerializer = ResponseSerializer<T, NSError> { request, response, data, error in
+            guard error == nil else { return .Failure(error!) }
+            
+            let JSONResponseSerializer = Request.JSONResponseSerializer(options: .AllowFragments)
+            let result = JSONResponseSerializer.serializeResponse(request, response, data, error)
+            
+            switch result {
+            case .Success(let value):
+                if let
+                    response = response,
```
Comment (jemmaSlater) [inline AlamofireRequest.swift]: is this step necessary?

File: AlamofireRequest.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                    responseObject = T(response: response, representation: value)
+                {
+                    return .Success(responseObject)
+                } else {
+                    let failureReason = "JSON could not be serialized into response object: \(value)"
+                    let error = Error.errorWithCode(.JSONSerializationFailed, failureReason: failureReason)
```
Comment (jemmaSlater) [inline AlamofireRequest.swift:38]: you could just put the failureReason string directly in here.. not sure which is more optimal.. probably just down to personal preference :)

File: AlamofireRequest.swift (hunk 1 above)
Comment (jemmaSlater) [inline AlamofireRequest.swift]: oh.. it checks its not nil?

File: AlamofireRequest.swift [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            let JSONSerializer = Request.JSONResponseSerializer(options: .AllowFragments)
+            let result = JSONSerializer.serializeResponse(request, response, data, error)
+            
+            switch result {
+            case .Success(let value):
+                if let response = response {
+                    return .Success(T.collection(response: response, representation: value))
+                } else {
+                    let failureReason = "Response collection could not be serialized due to nil response"
+                    let error = Error.errorWithCode(.JSONSerializationFailed, failureReason: failureReason)
```
Comment (jemmaSlater) [inline AlamofireRequest.swift:62]: again here could be one line? but up to you..

File: SharedConstants.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                static let bio = "bio"
+            }
+            
+            struct RibotName {
+                static let first = "first"
+                static let last = "last"
+            }
+        }
+    }
+}
```
Comment (jemmaSlater) [inline SharedConstants.swift]: no new line

File: SharedDataManager.swift [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            case .Failure(let error):
+                completionHandler(DataResult.Failure(error))
+            }
+        }
+    }
+
+    // MARK: - Shared Instance
+    
+    static let sharedInstance = SharedDataManager()
+}
```
Comment (jemmaSlater) [inline SharedDataManager.swift]: no new line

File: SharedRibotProfile.swift [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        self.email       = email
+        self.hexColor    = hexColor
+        self.ribotar     = ribotar
+        self.dateOfBirth = dateOfBirth
+    }
+    
+    // MARK: - Description
+    
+    override var description: String {
+        var description : String = self.name.description + "\n" + self.email + "\n" + self.hexColor + "\n" + self.ribotar.absoluteString + "\n" + self.dateOfBirth
```
Comment (jemmaSlater) [inline SharedRibotProfile.swift]: spacing? var description: String =

File: SharedRibotProfile.swift [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+// Mark: - Alamofire ResponseObjectSerializable implementation
+
+extension SharedRibotProfile: ResponseObjectSerializable, ResponseCollectionSerializable  {
+    
+    convenience init?(response: NSHTTPURLResponse, representation: AnyObject) {
+        if let ribotProfileRepresentation = representation.valueForKeyPath(SharedConstants.RibotAPI.Keys.Ribot.ribotProfile),
+            nameRepresentation = ribotProfileRepresentation.valueForKeyPath(SharedConstants.RibotAPI.Keys.RibotProfile.name),
+            email = ribotProfileRepresentation.valueForKeyPath(SharedConstants.RibotAPI.Keys.RibotProfile.email) as? String,
+            hexColor = ribotProfileRepresentation.valueForKeyPath(SharedConstants.RibotAPI.Keys.RibotProfile.hexColor) as? String,
+            ribotar = ribotProfileRepresentation.valueForKeyPath(SharedConstants.RibotAPI.Keys.RibotProfile.ribotar) as? String,
```
Comment (jemmaSlater) [inline SharedRibotProfile.swift]: does this work? Isn't ribotar an NSURL, not a String..?

File: AlamofireRequest.swift (hunk 1 above)
Comment (manuelmarcos) [inline AlamofireRequest.swift]: yeah! I just made some alignments :)

File: AlamofireRequest.swift (hunk 2 above)
Comment (manuelmarcos) [inline AlamofireRequest.swift:38]: yeah, it's just for the shake of readability :-)

**Comment by manuelmarcos** [comment]
Wicked! It's ready to review again :) Thanks Jemma!

**Pull request merged**